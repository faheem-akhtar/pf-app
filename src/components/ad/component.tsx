import { FunctionComponent } from 'react';

import styles from './ad.module.scss';

export const AdComponent: FunctionComponent<{ id: string }> = ({ id }) => <div data-ad className={styles.ad} id={id} />;
