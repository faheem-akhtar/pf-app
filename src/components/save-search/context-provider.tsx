import { FunctionComponent, useContext, useEffect, useState } from 'react';

import { apiSaveSearchCreateFetcher } from 'api/save-search/create/fetcher';
import { useApiSaveSearch } from 'api/save-search/hook';
import { FiltersContext } from 'components/filters/context';
import { UserContext } from 'components/user/context';
import { FiltersParametersEnum } from 'enums/filters/parameters.enum';

import { SaveSearchContext } from './context';
import { SaveSearchContextInterface } from './context.interface';
import { saveSearchFilterEquality } from './filter-equality';
import { saveSearchFiltersMapper } from './filters-mapper';
import { SaveSearchInterface } from './interface';
import { saveSearchTracker } from './tracker';

export const SaveSearchContextProvider: FunctionComponent = ({ children }) => {
  const user = useContext(UserContext);
  const saveSearchResponse = useApiSaveSearch();
  const saveSearchResponseData = saveSearchResponse.ok ? saveSearchResponse.data : [];
  const [searches, setSearches] = useState<SaveSearchInterface[]>([]);
  const filtersCtx = useContext(FiltersContext);
  const filters = saveSearchFiltersMapper(filtersCtx?.value || {});

  useEffect(() => {
    if (user && saveSearchResponse.ok) {
      setSearches(saveSearchResponseData);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [user, saveSearchResponse.ok]);

  const value: SaveSearchContextInterface = {
    ok: saveSearchResponse.ok,
    data: searches,
    filtered: searches.filter((item) => saveSearchFilterEquality(item.filters, filters)),
    create: ({ name, frequency }) =>
      apiSaveSearchCreateFetcher({ name, frequency, filters: filtersCtx.value }).then((response) => {
        if (response.ok) {
          setSearches([...searches, response.data]);
          saveSearchTracker.onCreateSuccess(
            filtersCtx.value[FiltersParametersEnum.locationsIds].map((location) => location.id)
          );
        }
        return response;
      }),
  };

  return <SaveSearchContext.Provider value={value}>{children}</SaveSearchContext.Provider>;
};
