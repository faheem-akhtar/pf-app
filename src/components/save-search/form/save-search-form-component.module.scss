.save-search-form__header {
  padding-bottom: var(--spacing);
}

.save-search-form__header-title {
  color: var(--color-global-black);
  font-size: var(--font-size-01);
  font-weight: var(--font-weight-02);
  line-height: 1.5;
}

.save-search-form__content-desc {
  color: var(--color-global-grey1);
  font-size: var(--font-size-08);
  font-weight: var(--font-weight-04);
  line-height: 1.5;
  margin-bottom: calc(var(--spacing) * 5);
}

.save-search-form__content-row {
  margin-bottom: calc(var(--spacing) * 5);
}

.save-search-form__label {
  color: var(--color-global-black);
  display: block;
  font-size: var(--font-size-08);
  font-weight: var(--font-weight-04);
  line-height: 1.5;
  margin-bottom: var(--spacing);
}

.save-search-form__input {
  border-radius: 1.6rem;
  padding: calc(var(--spacing) * 2) calc(var(--spacing) * 3);
  width: 100%;
}

.save-search-form__error--general {
  background-color: var(--color-feedback-error-lighter);
  border: 0.1rem solid var(--color-feedback-error-lighter);
  border-radius: 0.4rem;
  color: var(--color-primary-clay-100);
  display: inline-flex;
  font-size: var(--font-size-08);
  margin-bottom: 1rem;
  padding: calc(var(--spacing) * 1.33) calc(var(--spacing) * 3.3);
  width: 100%;
}

.save-search-form__error {
  color: var(--color-feedback-error-base);
  font-size: var(--font-size-08);
  font-weight: var(--font-weight-04);
  line-height: 1.5;
  padding-top: var(--spacing);
}

.save-search-form__info {
  color: var(--color-global-grey1);
  font-size: var(--font-size-08);
  font-weight: var(--font-weight-04);
  line-height: 1.5;
  padding-top: calc(var(--spacing) * 2);
}

.save-search-form__footer {
  bottom: 0;
  left: 0;
  padding: calc(var(--spacing) * 3);
  position: absolute;
  right: 0;
}

.save-search-form__button {
  width: 100%;
}

.save-search-form__button .dropdown__loader {
  height: 100%;
}

.save-search-form__content {
  overflow-y: auto;
  padding-bottom: calc(var(--spacing) * 9);
}
