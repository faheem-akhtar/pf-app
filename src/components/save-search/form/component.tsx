import { useContext, useState } from 'react';

import { formValidatorMaxLength } from 'helpers/form/validator/max-length';
import { formValidatorRequired } from 'helpers/form/validator/required';
import { saveSearchFrequencies } from '../frequencies';
import { useFormField } from 'helpers/form/field.hook';
import { useTranslation } from 'helpers/translation/hook';

import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { InputBaseComponent } from 'library/input/base/component';
import { SaveSearchContext } from '../context';
import { SaveSearchFrequencyEnum } from 'enums/save-search/frequency.enum';
import { SelectFieldTemplate } from 'library/select-field/template';

import styles from './save-search-form-component.module.scss';

const MAX_CHARACTERS_LIMIT = 256;

export const SaveSearchFormComponent = ({ onSuccess }: { onSuccess: () => void }): JSX.Element => {
  const { t } = useTranslation();
  const saveSearch = useContext(SaveSearchContext);
  const [generalError, setGeneralError] = useState<string>();
  const [isLoading, setIsloading] = useState(false);
  const [name, setName] = useFormField<string>('', [
    formValidatorRequired(t('save_search/name-validation-required')),
    formValidatorMaxLength(t('save_search/name-validation-character_limit'), MAX_CHARACTERS_LIMIT, true),
  ]);
  const [frequency, setFrequency] = useFormField<SaveSearchFrequencyEnum>(SaveSearchFrequencyEnum.DAILY);
  const fields = {
    id: null,
    name: '',
    formatted_filters: null,
    frequency: SaveSearchFrequencyEnum.DAILY,
  };
  const selectedFrequency = saveSearchFrequencies.find((item) => item.value === frequency.value);

  return (
    <>
      <header className={styles['save-search-form__header']}>
        <h1 className={styles['save-search-form__header-title']}>{t('save_search/form-add-title')}</h1>
      </header>
      <div className={styles['save-search-form__content']}>
        {generalError && <div className={styles['save-search-form__error--general']}>{generalError}</div>}
        {fields.formatted_filters && (
          <p className={styles['save-search-form__content-desc']}>{fields.formatted_filters}</p>
        )}
        <div className={styles['save-search-form__content-row']}>
          <label className={styles['save-search-form__label']}>{t('save_search/name-label')}</label>
          <InputBaseComponent
            floatPlaceholder={false}
            placeholder={t('save_search/name-placeholder')}
            value={name.value}
            error={!!name.error}
            onChange={(value): void => {
              setGeneralError('');
              setName(value);
            }}
          />
          {name.error && <p className={styles['save-search-form__error']}>{name.error}</p>}
        </div>
        <div className={styles['save-search-form__content-row']}>
          <label className={styles['save-search-form__label']} htmlFor='frequency'>
            {t('save_search/frequency-label')}
          </label>
          <SelectFieldTemplate
            value={selectedFrequency?.value}
            dropdownIcon
            options={saveSearchFrequencies.map((item) => ({ ...item, label: t(item.label) }))}
            onChange={(value): void => {
              setGeneralError('');
              setFrequency(value as SaveSearchFrequencyEnum);
            }}
          />
          {frequency.error && <p className={styles['save-search-form__error']}>{frequency.error}</p>}
          <p className={styles['save-search-form__info']}>{t('save_search/frequency-help_text')}</p>
        </div>
      </div>
      <footer className={styles['save-search-form__footer']}>
        <ButtonTemplate
          type='button'
          size={ButtonSizeEnum.regular}
          className={styles['save-search-form__button']}
          componentType={ButtonComponentTypeEnum.primary}
          onClick={(): void => {
            const searchName = name.value?.trim();
            if (!searchName || name.error) {
              setName(searchName);
              return;
            }
            setIsloading(true);
            saveSearch
              .create({
                name: searchName,
                frequency: frequency.value,
              })
              .then((response) => {
                setIsloading(false);
                if (response.ok) {
                  onSuccess();
                } else {
                  setGeneralError(t('general-error-message'));
                }
              });
          }}
          disabled={isLoading}
          loading={isLoading}
        >
          {t('save')}
        </ButtonTemplate>
      </footer>
    </>
  );
};
