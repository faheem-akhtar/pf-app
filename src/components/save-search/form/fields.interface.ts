import { BackendModelSaveSearchInterface } from 'backend/model/save-search/interface';

export interface SaveSearchFormFieldsInterface
  extends Pick<BackendModelSaveSearchInterface, 'id' | 'name' | 'frequency' | 'formatted_filters'> {}
