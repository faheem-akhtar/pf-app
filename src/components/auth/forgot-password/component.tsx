import React, { useState } from 'react';

import { AuthForgotPasswordFieldEnum } from 'components/auth/forgot-password/field.enum';
import { AuthForgotPasswordPropsInterface } from 'components/auth/forgot-password/props.interface';
import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { AuthResetPasswordService } from 'services/auth/reset-password.service';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ErrorMessageComponent } from 'components/error-message/component';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google/recaptcha.service';
import { InputBaseComponent } from 'library/input/base/component';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import styles from '../auth.module.scss';

export const AuthForgotPasswordComponent = (props: AuthForgotPasswordPropsInterface): JSX.Element => {
  const captchaService = GoogleRecaptchaService();
  const { t } = useTranslation();
  const [email, setEmail] = useState('');
  const [errors, setErrors] = useState<Partial<Record<AuthLoginFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthLoginFieldEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
  }));
  const [isLoading, setIsLoading] = useState(false);

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(errors, setErrors, validators);

  // TODO-FE[CX-499] - Add success message
  return (
    <div data-testid='AuthForgotPasswordComponent'>
      <div className={styles.heading}>{t('auth/forgot-password')}?</div>
      <AuthLoaderComponent isEnabled={isLoading} isCentered={true} />
      {errorMessage && <ErrorMessageComponent message={errorMessage} />}
      <form
        onSubmit={(e): void => {
          e.preventDefault();

          if (
            !validate({
              [AuthForgotPasswordFieldEnum.email]: email,
            })
          ) {
            setIsLoading(true);
            captchaService.execute().then((captcha_token) => {
              AuthResetPasswordService({
                email,
                captcha_token,
              })
                .then((e) => {
                  if (e.ok) {
                    // Reset server error message
                    setErrorMessage('');
                    // Close modal
                    props.onClose();
                    props.onSuccess();
                  } else {
                    setErrorMessage(e.error?.body || `${t('auth/something-wrong')}! ${t('auth/try-later')}`);
                    captchaService.reset();
                  }
                })
                .finally(() => setIsLoading(false));
            });
          }
          return;
        }}
      >
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('email')}
            type='email'
            value={email}
            error={!!errors[AuthForgotPasswordFieldEnum.email]}
            onChange={(value): void => {
              validate({
                [AuthForgotPasswordFieldEnum.email]: value,
              });
              setEmail(value);
            }}
            errorText={errors[AuthForgotPasswordFieldEnum.email]}
          />
        </div>
        <div className={styles['input-area']}>
          <ButtonTemplate
            className={styles.button}
            type='submit'
            componentType={ButtonComponentTypeEnum.primary}
            size={ButtonSizeEnum.regular}
          >
            {t('auth/reset-password')}
          </ButtonTemplate>
        </div>
        <ReCaptchaComponent />
      </form>
      <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
        {t('auth/not-registered-yet')}?&nbsp;{t('auth/create-account')}
      </div>
    </div>
  );
};
