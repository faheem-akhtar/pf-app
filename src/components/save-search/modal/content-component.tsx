import { useRouter } from 'next/router';
import { useContext } from 'react';

import { IconThickCloseTemplate } from 'components/icon/thick/close-template';
import { SnackbarContext } from 'components/snackbar/context';
import { localeGetHref } from 'helpers/locale/get-href';
import { useTranslation } from 'helpers/translation/hook';

import { SaveSearchFormComponent } from '../form/component';
import styles from './save-search-modal-component.module.scss';

export const SaveSearchModalContentComponent = ({ close }: { close: () => void }): JSX.Element => {
  const snackbar = useContext(SnackbarContext);
  const locale = useRouter().locale as string;
  const { t } = useTranslation();

  return (
    <div className={styles.content}>
      <SaveSearchFormComponent
        onSuccess={(): void => {
          snackbar.alert({
            message: t('save_search/create-success_notification'),
            action: (
              <a href={localeGetHref(locale, '/user/saved-searches', true)}>{t('save_search/manage-cta-label')}</a>
            ),
          });
          close();
        }}
      />
      <div className={styles.close} onClick={close}>
        <IconThickCloseTemplate class={styles.closeIcon} />
      </div>
    </div>
  );
};
