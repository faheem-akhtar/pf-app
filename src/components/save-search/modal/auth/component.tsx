import React, { FunctionComponent, useContext, useEffect, useRef } from 'react';

import { AuthModalComponent } from 'components/auth/modal/component';
import { ModalComponent } from 'components/modal/component';
import { SaveSearchContext } from 'components/save-search/context';
import { saveSearchTracker } from 'components/save-search/tracker';
import { UserContext } from 'components/user/context';
import { functionNoop } from 'helpers/function/noop';
import { AuthService } from 'services/auth/service';
import { AuthSubscribeEventTypeEnum } from 'services/auth/subscribe-event-type.enum';
import { AuthSubscriberType } from 'services/auth/subscriber.type';

import { SaveSearchModalAuthPropsInterface } from './props.interface';

export const SaveSearchModalAuthComponent: FunctionComponent<SaveSearchModalAuthPropsInterface> = ({
  onCancel,
  onSuccess,
}) => {
  const user = useContext(UserContext);
  const saveSearch = useContext(SaveSearchContext);

  const openLoginRef = useRef(functionNoop);
  const closeLoginRef = useRef(functionNoop);

  useEffect(() => {
    const onUpdate: AuthSubscriberType = (newUser, meta) => {
      if (newUser && meta?.eventType === AuthSubscribeEventTypeEnum.login) {
        saveSearchTracker.onSignInSuccess();
      }
      if (newUser && meta?.eventType === AuthSubscribeEventTypeEnum.register) {
        saveSearchTracker.onSignUpSuccess();
      }
    };
    const unsubscribeAuth = AuthService.subscribe(onUpdate);

    openLoginRef.current();

    return (): void => unsubscribeAuth();
  });

  useEffect(() => {
    if (user && saveSearch.ok) {
      onSuccess();
    }
  }, [user, saveSearch.ok, onSuccess]);

  return (
    <ModalComponent openRef={openLoginRef} closeRef={closeLoginRef} overlay>
      <AuthModalComponent cancel={onCancel} />
    </ModalComponent>
  );
};
