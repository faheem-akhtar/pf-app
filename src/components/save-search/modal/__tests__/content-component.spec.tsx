/**
 * @jest-environment jsdom
 */

import { act, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import { mockSnackbarEnv } from 'mocks/snackbar/env.mock';

import { SnackbarContextProvider } from 'components/snackbar/context-provider';

import * as SaveSearchFormComponentModule from '../../form/component';
import { SaveSearchModalContentComponent } from '../content-component';

describe('SaveSearchModalContentComponent', () => {
  beforeEach(() => {
    mockSnackbarEnv();
  });

  it('should render the component', () => {
    expect(render(<SaveSearchModalContentComponent close={jest.fn} />)).toMatchSnapshot();
  });

  it('should display message on success', async () => {
    const closeMock = jest.fn();
    jest
      .spyOn(SaveSearchFormComponentModule, 'SaveSearchFormComponent')
      .mockImplementationOnce(({ onSuccess }) => <button onClick={onSuccess} />);

    const { getByRole, getByText } = render(
      <SnackbarContextProvider>
        <SaveSearchModalContentComponent close={closeMock} />
      </SnackbarContextProvider>
    );

    act(() => {
      userEvent.click(getByRole('button'));
    });

    await waitFor(() => getByText('save_search/create-success_notification'));

    expect(closeMock).toHaveBeenCalledTimes(1);
    expect(closeMock).toHaveBeenCalledWith();
  });
});
