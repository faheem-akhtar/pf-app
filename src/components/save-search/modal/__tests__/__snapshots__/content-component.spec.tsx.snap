// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveSearchModalContentComponent should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="snackbar-root"
    />
    <div
      id="root_element"
    />
    <div>
      <div
        class="content"
      >
        <header
          class="save-search-form__header"
        >
          <h1
            class="save-search-form__header-title"
          >
            save_search/form-add-title
          </h1>
        </header>
        <div
          class="save-search-form__content"
        >
          <div
            class="save-search-form__content-row"
          >
            <label
              class="save-search-form__label"
              for="save-search-name"
            >
              save_search/name-label
            </label>
            <div
              class="container"
            >
              <div
                class="content"
              >
                <input
                  aria-invalid="false"
                  aria-label="name"
                  class="input"
                  id="save-search-name"
                  name="name"
                  placeholder="save_search/name-placeholder"
                  type="text"
                  value=""
                />
              </div>
            </div>
            
          </div>
          <div
            class="save-search-form__content-row"
          >
            <label
              class="save-search-form__label"
              for="frequency"
            >
              save_search/frequency-label
            </label>
            <div
              class="container"
            >
              <div
                class="fieldWrapper"
              >
                <select
                  class="field fieldWithIcon"
                  id="frequency"
                  name="frequency"
                >
                  <option
                    value="0"
                  >
                    off
                  </option>
                  <option
                    value="1"
                  >
                    daily
                  </option>
                  <option
                    value="2"
                  >
                    weekly
                  </option>
                </select>
                <svg
                  class="icon"
                  viewBox="4.5 8 16 8"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.64645 8.14645C4.47288 8.32001 4.4536 8.58944 4.58859 8.78431L4.64645 8.85355L11.6464 15.8536C11.82 16.0271 12.0894 16.0464 12.2843 15.9114L12.3536 15.8536L19.3536 8.85355C19.5488 8.65829 19.5488 8.34171 19.3536 8.14645C19.18 7.97288 18.9106 7.9536 18.7157 8.08859L18.6464 8.14645L12 14.793L5.35355 8.14645C5.17999 7.97288 4.91056 7.9536 4.71569 8.08859L4.64645 8.14645Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
            <p
              class="save-search-form__info"
            >
              save_search/frequency-help_text
            </p>
          </div>
        </div>
        <footer
          class="save-search-form__footer"
        >
          <button
            class="button primary regular save-search-form__button"
          >
            save
          </button>
        </footer>
        <div
          class="close"
        >
          <svg
            class="closeIcon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <title>
              Close
            </title>
            <path
              clip-rule="evenodd"
              d="M5.6129 4.2097L5.70711 4.29289L12 10.585L18.2929 4.29289C18.6834 3.90237 19.3166 3.90237 19.7071 4.29289C20.0676 4.65338 20.0953 5.22061 19.7903 5.6129L19.7071 5.70711L13.414 12L19.7071 18.2929C20.0976 18.6834 20.0976 19.3166 19.7071 19.7071C19.3466 20.0676 18.7794 20.0953 18.3871 19.7903L18.2929 19.7071L12 13.414L5.70711 19.7071C5.31658 20.0976 4.68342 20.0976 4.29289 19.7071C3.93241 19.3466 3.90468 18.7794 4.2097 18.3871L4.29289 18.2929L10.585 12L4.29289 5.70711C3.90237 5.31658 3.90237 4.68342 4.29289 4.29289C4.62334 3.96245 5.12751 3.91161 5.5114 4.14038L5.6129 4.2097Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content"
    >
      <header
        class="save-search-form__header"
      >
        <h1
          class="save-search-form__header-title"
        >
          save_search/form-add-title
        </h1>
      </header>
      <div
        class="save-search-form__content"
      >
        <div
          class="save-search-form__content-row"
        >
          <label
            class="save-search-form__label"
            for="save-search-name"
          >
            save_search/name-label
          </label>
          <div
            class="container"
          >
            <div
              class="content"
            >
              <input
                aria-invalid="false"
                aria-label="name"
                class="input"
                id="save-search-name"
                name="name"
                placeholder="save_search/name-placeholder"
                type="text"
                value=""
              />
            </div>
          </div>
          
        </div>
        <div
          class="save-search-form__content-row"
        >
          <label
            class="save-search-form__label"
            for="frequency"
          >
            save_search/frequency-label
          </label>
          <div
            class="container"
          >
            <div
              class="fieldWrapper"
            >
              <select
                class="field fieldWithIcon"
                id="frequency"
                name="frequency"
              >
                <option
                  value="0"
                >
                  off
                </option>
                <option
                  value="1"
                >
                  daily
                </option>
                <option
                  value="2"
                >
                  weekly
                </option>
              </select>
              <svg
                class="icon"
                viewBox="4.5 8 16 8"
              >
                <path
                  clip-rule="evenodd"
                  d="M4.64645 8.14645C4.47288 8.32001 4.4536 8.58944 4.58859 8.78431L4.64645 8.85355L11.6464 15.8536C11.82 16.0271 12.0894 16.0464 12.2843 15.9114L12.3536 15.8536L19.3536 8.85355C19.5488 8.65829 19.5488 8.34171 19.3536 8.14645C19.18 7.97288 18.9106 7.9536 18.7157 8.08859L18.6464 8.14645L12 14.793L5.35355 8.14645C5.17999 7.97288 4.91056 7.9536 4.71569 8.08859L4.64645 8.14645Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <p
            class="save-search-form__info"
          >
            save_search/frequency-help_text
          </p>
        </div>
      </div>
      <footer
        class="save-search-form__footer"
      >
        <button
          class="button primary regular save-search-form__button"
        >
          save
        </button>
      </footer>
      <div
        class="close"
      >
        <svg
          class="closeIcon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <title>
            Close
          </title>
          <path
            clip-rule="evenodd"
            d="M5.6129 4.2097L5.70711 4.29289L12 10.585L18.2929 4.29289C18.6834 3.90237 19.3166 3.90237 19.7071 4.29289C20.0676 4.65338 20.0953 5.22061 19.7903 5.6129L19.7071 5.70711L13.414 12L19.7071 18.2929C20.0976 18.6834 20.0976 19.3166 19.7071 19.7071C19.3466 20.0676 18.7794 20.0953 18.3871 19.7903L18.2929 19.7071L12 13.414L5.70711 19.7071C5.31658 20.0976 4.68342 20.0976 4.29289 19.7071C3.93241 19.3466 3.90468 18.7794 4.2097 18.3871L4.29289 18.2929L10.585 12L4.29289 5.70711C3.90237 5.31658 3.90237 4.68342 4.29289 4.29289C4.62334 3.96245 5.12751 3.91161 5.5114 4.14038L5.6129 4.2097Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
